types/rethinkdb/index.d.ts

```diff
@@ -523,9 +574,39 @@ declare module "rethinkdb" {
 
     interface Expression<T> extends Writeable, Operation<T>, HasFields<Expression<number>> {
         (prop: string): Expression<any>;
+
+        /**
+         * merge([object | function, object | function, ...])
+         *
+         * Merge two or more objects together to construct a new object with properties from all.
+         * When there is a conflict between field names, preference is given to fields in the
+         * rightmost object in the argument list. merge also accepts a subquery function that
+         * returns an object, which will be used similarly to a map function.
+         *
+         * Example: Equip Thor for battle.
+         *
+         * r.table('marvel').get('thor').merge(
+         *     r.table('equipment').get('hammer'),
+         *     r.table('equipment').get('pimento_sandwich')
+         * ).run(conn, callback)
+         */
         merge(query: Expression<Object>): Expression<Object>;
-        append(prop: string): Expression<Object>;
-        contains(prop: string): Expression<boolean>;
+        merge(object: Object): Expression<Object>;
+        merge(cb: ExpressionFunction<Expression<any>>): Expression<Object>;
+
+        append(element: any | Expression<any>): Expression<Object>;
+        /**
+         * contains([value | predicate_function, ...])
+         *
+         * When called with values, returns true if a sequence contains all the specified values.
+         * When called with predicate functions, returns true if for each predicate there exists
+         * at least one element of the stream where that predicate returns true.
+         *
+         * Example: Has Iron Man ever fought Superman?
+         *
+         * r.table('marvel').get('ironman')('opponents').contains('superman').run(conn, callback)
+         */
+        contains(element: any | Expression<any> | ExpressionFunction<boolean>): Expression<boolean>;
```

Comment:
As far as I know, `any | something else` just means `any` and the rest does nothing